ts/store/reducers/entities/services/currentService.ts

```diff
@@ -0,0 +1,36 @@
+import { NavigationActions } from "react-navigation";
+import { createSelector } from "reselect";
+import { Action } from "../../../actions/types";
+import ROUTES from "../../../../navigation/routes";
+import { GlobalState } from "../../types";
+import { ServiceID } from "../../../../types/services/ServicePreference";
+
+export type CurrentServiceSelectedState = {
+  serviceID: ServiceID;
+};
+
+export const currentServiceSelectedReducer = (
+  state: CurrentServiceSelectedState | null = null,
+  action: Action
+): CurrentServiceSelectedState | null => {
+  switch (action.type) {
+    case NavigationActions.NAVIGATE:
```

Comment:
imho a new explicit action should be created for this scope, e.g `currentService`

```typescript
export const currentSelectedService = createStandardAction(
  "CURRENT_SELECTED_SERVICE"
)<ServiceID>();
```
it could be dispatched just before where `ROUTES.SERVICE_DETAIL` is dispatched (if FF is enabled)

changing the state using a not explicit action could be dangerous: we know that a service is selected because a specific navigation action is dispatched (this is an implicit/derived info)

furthermore this reducer evaluates the statement `if(action.routeName === ROUTES.SERVICE_DETAIL)` when every navigation action is dispatched